File: iOS4bookExamples/p740p758XML/p740XML/MyPersonParser.m


#import "MyPersonParser.h"
#import "Person.h"

@implementation MyPersonParser
@synthesize person;

- (void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qualifiedName attributes:(NSDictionary *)attributeDict {
    [self makeChild:[MyXMLParserDelegate class] elementName:elementName parser:parser];
}

- (void) finishedChild:(NSString *)s {
    if (!self.person) {
        Person* p = [[Person alloc] init];
        self.person = p;
        [p release];
    }
    [self.person setValue: s forKey: self.child.name];
}

- (void) dealloc {
    [person release];
    [super dealloc];
}


@end

File: iOS10bookExampless/bk2ch23p818xml/ch36p1070xml/MyPersonParser.swift


import UIKit

class MyPersonParser : MyXMLParserDelegate {
    
    var person = Person(firstName: "", lastName: "")
    
    func parser(_ parser: XMLParser, didStartElement elementName: String,
        namespaceURI: String?, qualifiedName qName: String?,
        attributes attributeDict: [String : String]) {
            self.makeChild(MyXMLParserDelegate.self, elementName: elementName, parser: parser)
    }
    
    override func finishedChild(_ s: String) {
        self.person.setValue(s, forKey:self.child.name)
    }
    
}
